lib/demos/reference/motion_demo_shared_x_axis_transition.dart

```diff
@@ -0,0 +1,249 @@
+                    disabledColor: Colors.black12,
+                    child: Text(localizations.demoSharedXAxisNextButtonText),
+                  ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _CoursePage extends StatelessWidget {
+  const _CoursePage();
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+
+    return ListView(
+      children: [
+        const Padding(
+          padding: EdgeInsets.symmetric(vertical: 8),
+        ),
+        Text(
+          localizations.demoSharedXAxisCoursePageTitle,
+          style: Theme.of(context).textTheme.headline5,
+          textAlign: TextAlign.center,
+        ),
+        const Padding(
+          padding: EdgeInsets.symmetric(vertical: 5),
+        ),
+        Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 10),
+          child: Text(
+            localizations.demoSharedXAxisCoursePageSubtitle,
+            style: const TextStyle(
+              fontSize: 12,
+              color: Colors.grey,
+            ),
+            textAlign: TextAlign.center,
+          ),
+        ),
+        _CourseSwitch(
+            course: localizations.demoSharedXAxisArtsAndCraftsCourseTitle),
+        _CourseSwitch(course: localizations.demoSharedXAxisBusinessCourseTitle),
+        _CourseSwitch(
+            course: localizations.demoSharedXAxisIllustrationCourseTitle),
+        _CourseSwitch(course: localizations.demoSharedXAxisDesignCourseTitle),
+        _CourseSwitch(course: localizations.demoSharedXAxisCulinaryCourseTitle),
+      ],
+    );
+  }
+}
+
+class _CourseSwitch extends StatefulWidget {
+  const _CourseSwitch({
+    this.course,
+  });
+
+  final String course;
+
+  @override
+  _CourseSwitchState createState() => _CourseSwitchState();
+}
+
+class _CourseSwitchState extends State<_CourseSwitch> {
+  bool _value = true;
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+    final subtitle = _value
+        ? localizations.demoSharedXAxisBundledCourseSubtitle
+        : localizations.demoSharedXAxisIndividualCourseSubtitle;
+
+    return SwitchListTile(
+      title: Text(widget.course),
+      subtitle: Text(subtitle),
+      value: _value,
+      onChanged: (newValue) {
+        setState(() {
+          _value = newValue;
+        });
+      },
+    );
+  }
+}
+
+class _SignInPage extends StatelessWidget {
+  const _SignInPage();
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        final maxHeight = constraints.maxHeight;
+
+        return Column(
+          children: [
+            Padding(
+              padding: EdgeInsets.symmetric(vertical: maxHeight / 20),
+            ),
```

Comment:
You can just extract this into a variable below maxHeight declaration and use SizedBox instead, then you can reuse it everywhere here

```
final spacing = SizedBox(height: maxHeight/ 10);
```